### PR TITLE
Add azurerm_resource_group resource

### PR DESCRIPTION
**Pull request by sundowndev**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #1029 
| ❓ Documentation  | https://github.com/cloudskiff/driftctl-docs/pull/128 

## Description

This PR implements the `azurerm_resource_group` resource type.

### CODE REVIEW

File: pkg/remote/azurerm/init.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func Init(
 	}
 
 	providerConfig := provider.GetConfig()
-	cred, err := azidentity.NewDefaultAzureCredential(&azidentity.DefaultAzureCredentialOptions{})
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
```
Comment (eliecharra) [inline pkg/remote/azurerm/init.go:40]: I don't think this should be changed

File: pkg/resource/azurerm/azurerm_resource_group.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import "github.com/cloudskiff/driftctl/pkg/resource"
+
+const AzureResourceGroupResourceType = "azurerm_resource_group"
+
+func initAzureResourceGroupMetadata(resourceSchemaRepository resource.SchemaRepositoryInterface) {
+	resourceSchemaRepository.SetNormalizeFunc(AzureResourceGroupResourceType, func(res *resource.Resource) {
+		val := res.Attrs
+		val.SafeDelete([]string{"timeouts"})
+	})
```
Comment (eliecharra) [inline pkg/resource/azurerm/azurerm_resource_group.go:11]: We do not support deep mode for this resource, this code is useless

File: pkg/resource/azurerm/azurerm_resource_group_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+func TestAcc_Azure_ResourceGroup(t *testing.T) {
+	acceptance.Run(t, acceptance.AccTestCase{
+		TerraformVersion: "0.15.5",
+		Paths:            []string{"./testdata/acc/azurerm_resource_group"},
+		Args: []string{
+			"scan",
+			"--to", "azure+tf",
+			"--filter", "Type=='azurerm_resource_group' && contains(Id, 'acc-test-res-group-')",
```
Comment (eliecharra) [inline pkg/resource/azurerm/azurerm_resource_group_test.go:17]: Nice trick

File: pkg/remote/azurerm/repository/armresources.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/Azure/azure-sdk-for-go/sdk/resources/armresources"
+	"github.com/cloudskiff/driftctl/pkg/remote/azurerm/common"
+	"github.com/cloudskiff/driftctl/pkg/remote/cache"
+)
+
+type ArmResourcesRespository interface {
+	ListAllResourceGroups() ([]*armresources.ResourceGroup, error)
+}
+
+type armResourcesListPager interface {
```
Comment (eliecharra) [inline pkg/remote/azurerm/repository/armresources.go:16]: You can compose the pager interface I added in virtual network PR, we'll need to merge mine first and then rebase this one.

```go

// pager.go
type pager interface {
	Err() error
	NextPage(ctx context.Context) bool
}

// armresource.go
type armResourcesListPager interface {
    pager
	PageResponse() armresources.ResourceGroupsListResponse
}
```

File: pkg/remote/azurerm/repository/armresources.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resources/armresources"
+	"github.com/cloudskiff/driftctl/pkg/remote/azurerm/common"
+	"github.com/cloudskiff/driftctl/pkg/remote/cache"
+)
+
+type ArmResourcesRespository interface {
```
Comment (eliecharra) [inline pkg/remote/azurerm/repository/armresources.go:12]: You can remove `Arm` in naming to be consistent with other namings

File: pkg/remote/azurerm/repository/armresources.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+type armResourcesRepository struct {
+	client armResourcesClient
+	cache  cache.Cache
+}
+
+func NewArmResourcesRepository(con *arm.Connection, config common.AzureProviderConfig, cache cache.Cache) *armResourcesRepository {
+	return &armResourcesRepository{
+		armResourcesClientImpl{armresources.NewResourceGroupsClient(con, config.SubscriptionID)},
```
Comment (eliecharra) [inline pkg/remote/azurerm/repository/armresources.go:41]: `&armResourcesClientImpl`

**Review by eliecharra — CHANGES_REQUESTED** [review]

File: pkg/remote/azurerm/repository/armresources.go (hunk 4 above)
Comment (eliecharra) [reply pkg/remote/azurerm/repository/armresources.go:16]: You can do the rebase @sundowndev

File: pkg/remote/azurerm/init.go (hunk 1 above)
Comment (sundowndev) [reply pkg/remote/azurerm/init.go:40]: This is from the official docs https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/new-version-quickstart.md#authentication

Comment (eliecharra) [reply pkg/remote/azurerm/init.go]: 🆗

**Review by eliecharra — DISMISSED** [review]
👍🏻 Rebase needed

**Review by eliecharra — APPROVED** [review]

**Pull request merged**